File: Phase-1-Foundations/Python/Day020_LoopsRecap.py
```python
"""
Headline: Revision - Loops Recap
Description: Read a number and find its digit count, digit sum,
reverse, and whether it is a palindrome.

Pseudocode:

    num = int(input())

    digit_count = count_digits(num)
    digit_sum = sum_of_digits(num)
    reverse = reverse_number(num)
    palindrome = is_palindrome(num)

    print all results
"""


# ---- Functions ----

# Function 1: Count the number of digits
def count_digits(num):

    num = abs(num)

    if num == 0:
        return 1

    digit_count = 0

    while num != 0:
        digit_count = digit_count + 1
        num = num // 10

    return digit_count


# Function 2: Calculate the sum of digits
def sum_of_digits(num):

    num = abs(num)

    digit_sum = 0

    while num != 0:
        digit = num % 10
        digit_sum = digit_sum + digit
        num = num // 10

    return digit_sum


# Function 3: Reverse the number
def reverse_number(num):

    num = abs(num)

    reverse = 0

    while num != 0:
        digit = num % 10
        reverse = reverse * 10 + digit
        num = num // 10

    return reverse


# Function 4: Check whether the number is palindrome
def is_palindrome(num):

    # Negative numbers are not considered palindrome
    if num < 0:
        return False

    reverse = reverse_number(num)

    return num == reverse


# ---- Main Program ----

N = int(input("Enter a number: "))

digit_count = count_digits(N)
digit_sum = sum_of_digits(N)
reverse = reverse_number(N)
palindrome = is_palindrome(N)

print("Number of digits:", digit_count)
print("Sum of digits:", digit_sum)
print("Reversed number:", reverse)

if palindrome:
    print("Palindrome: Yes")
else:
    print("Palindrome: No")
```

File: Phase-1-Foundations/Java/Day020_LoopsRecap.java
```java
package Phase_1_foundations;

import java.util.Scanner;

/**
 * Headline: Revision - Loops Recap
 * Description: Read a number and find its digit count, digit sum,
 * reverse, and whether it is a palindrome.
 *
 * Pseudocode:
 *
 *     input(num);
 *
 *     digitCount = countDigits(num);
 *     digitSum = sumOfDigits(num);
 *     reverse = reverseNumber(num);
 *     palindrome = isPalindrome(num);
 *
 *     print all results;
 */

// ---- Functions ----

public class Day020_LoopsRecap {

    // Function 1: Count the number of digits
    public static int countDigits(int num) {

        num = Math.abs(num);

        if (num == 0) {
            return 1;
        }

        int digitCount = 0;

        while (num != 0) {
            digitCount++;
            num = num / 10;
        }

        return digitCount;
    }

    // Function 2: Calculate the sum of digits
    public static int sumOfDigits(int num) {

        num = Math.abs(num);

        int digitSum = 0;

        while (num != 0) {
            int digit = num % 10;
            digitSum = digitSum + digit;
            num = num / 10;
        }

        return digitSum;
    }

    // Function 3: Reverse the number
    public static int reverseNumber(int num) {

        num = Math.abs(num);

        int reverse = 0;

        while (num != 0) {
            int digit = num % 10;
            reverse = reverse * 10 + digit;
            num = num / 10;
        }

        return reverse;
    }

    // Function 4: Check whether the number is palindrome
    public static boolean isPalindrome(int num) {

        // Negative numbers are not considered palindrome
        if (num < 0) {
            return false;
        }

        int reverse = reverseNumber(num);

        return num == reverse;
    }

    // ---- Main Program ----

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a number: ");
        int num = sc.nextInt();

        int digitCount = countDigits(num);
        int digitSum = sumOfDigits(num);
        int reverse = reverseNumber(num);
        boolean palindrome = isPalindrome(num);

        System.out.println("Number of digits: " + digitCount);
        System.out.println("Sum of digits: " + digitSum);
        System.out.println("Reversed number: " + reverse);

        if (palindrome) {
            System.out.println("Palindrome: Yes");
        } else {
            System.out.println("Palindrome: No");
        }

        sc.close();
    }
}
```

